src/SourceLocator/Type/DirectorySourceLocator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace BetterReflection\SourceLocator\Type;
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+use BetterReflection\Reflector\Reflector;
+use BetterReflection\SourceLocator\Exception\InvalidDirectory;
+
+/**
+ * This source locator loads all php files in an entire directory or multiple directories.
+ */
+class DirectorySourceLocator implements SourceLocator
+{
+    /**
+     * @var AggregateSourceLocator
+     */
+    private $aggregatedSourceLocator;
+
+    /**
+     * @param $directories string[] directories to scan
+     */
+    public function __construct(array $directories)
+    {
+        $sourceLocators = [];
+        foreach ($directories as $dir) {
+            if (!is_string($dir)) {
+                throw InvalidDirectory::fromNonStringValue($dir);
+            } elseif (!is_dir($dir)) {
```

Comment:
but if it's a string and it's not a valid directory. the else if clause will handle this case.